internal/controllers/machine/drain/drain_test.go

```diff
@@ -1356,6 +1441,202 @@ func TestEvictPods(t *testing.T) {
+						},
+					},
+					Status: PodDeleteStatus{
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+						DrainOrder:    ptr.To[int32](-5), // min DrainOrder is -5 => will be evicted now
+						Reason:        PodDeleteStatusTypeOkay,
+					},
+				},
+				{
+					Pod: &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pod-3-to-trigger-eviction-successfully",
+						},
+					},
+					Status: PodDeleteStatus{
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+						DrainOrder:    ptr.To[int32](-5), // min DrainOrder is -5 => will be evicted now
+						Reason:        PodDeleteStatusTypeOkay,
+					},
+				},
+				{
+					Pod: &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pod-4-to-trigger-eviction-pod-not-found",
+						},
+					},
+					Status: PodDeleteStatus{
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+						DrainOrder:    ptr.To[int32](-5), // min DrainOrder is -5 => will be evicted now
+						Reason:        PodDeleteStatusTypeOkay,
+					},
+				},
+				{
+					Pod: &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pod-5-to-trigger-eviction-pdb-violated-2",
+						},
+					},
+					Status: PodDeleteStatus{
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+						DrainOrder:    ptr.To[int32](-5), // min DrainOrder is -5 => will be evicted now
+						Reason:        PodDeleteStatusTypeOkay,
+					},
+				},
+				{
+					Pod: &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pod-5-to-trigger-eviction-pdb-violated-1",
+						},
+					},
+					Status: PodDeleteStatus{
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+						DrainOrder:    ptr.To[int32](-5), // min DrainOrder is -5 => will be evicted now
+						Reason:        PodDeleteStatusTypeOkay,
+					},
+				},
+				{
+					Pod: &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pod-6-to-trigger-eviction-namespace-terminating",
+						},
+					},
+					Status: PodDeleteStatus{
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+						DrainOrder:    ptr.To[int32](-5), // min DrainOrder is -5 => will be evicted now
+						Reason:        PodDeleteStatusTypeOkay,
+					},
+				},
+				{
+					Pod: &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pod-7-to-trigger-eviction-some-other-error",
+						},
+					},
+					Status: PodDeleteStatus{
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+						DrainOrder:    ptr.To[int32](-5), // min DrainOrder is -5 => will be evicted now
+						Reason:        PodDeleteStatusTypeOkay,
+					},
+				},
+				{
+					Pod: &corev1.Pod{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "pod-8-to-trigger-eviction-later",
+						},
+					},
+					Status: PodDeleteStatus{
+						DrainBehavior: clusterv1.MachineDrainRuleDrainBehaviorDrain,
+						DrainOrder:    ptr.To[int32](6), // min DrainOrder is 6 => will be evicted later
```

Comment:
```suggestion
						DrainOrder:    ptr.To[int32](6), // min DrainOrder is -5 => will be evicted later
```
nit: this is referring to the calculated min order of all Pods